x-pack/filebeat/input/azureeventhub/v1_input.go

```diff
@@ -0,0 +1,359 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build !aix
+
+package azureeventhub
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/Azure/go-autorest/autorest/azure"
+
+	eventhub "github.com/Azure/azure-event-hubs-go/v3"
+	"github.com/Azure/azure-event-hubs-go/v3/eph"
+	"github.com/Azure/azure-event-hubs-go/v3/storage"
+	"github.com/Azure/azure-storage-blob-go/azblob"
+
+	v2 "github.com/elastic/beats/v7/filebeat/input/v2"
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+)
+
+// eventHubInputV1 is the Azure Event Hub input V1.
+//
+// This input uses the Azure Event Hub SDK v3 (legacy).
+type eventHubInputV1 struct {
+	config         azureInputConfig
+	log            *logp.Logger
+	metrics        *inputMetrics
+	processor      *eph.EventProcessorHost
+	pipelineClient beat.Client
+}
+
+// newEventHubInputV1 creates a new instance of the Azure Event Hub input V1.
+// This input uses the Azure Event Hub SDK v3 (legacy).
+func newEventHubInputV1(config azureInputConfig, logger *logp.Logger) (v2.Input, error) {
+	log := logger.
+		Named(inputName).
+		With(
+			"connection string", stripConnectionString(config.ConnectionString),
+		)
+
+	return &eventHubInputV1{
+		config: config,
+		log:    log,
+	}, nil
+}
+
+func (in *eventHubInputV1) Name() string {
+	return inputName
+}
+
+func (in *eventHubInputV1) Test(v2.TestContext) error {
+	return nil
+}
+
+func (in *eventHubInputV1) Run(
+	inputContext v2.Context,
+	pipeline beat.Pipeline,
+) error {
+	var err error
+
+	// Create pipelineClient for publishing events.
+	in.pipelineClient, err = createPipelineClient(pipeline)
+	if err != nil {
+		return fmt.Errorf("failed to create pipeline pipelineClient: %w", err)
+	}
+	defer in.pipelineClient.Close()
+
+	// Setup input metrics
+	in.metrics = newInputMetrics(inputContext.ID, nil)
+	defer in.metrics.Close()
+
+	ctx := v2.GoContextFromCanceler(inputContext.Cancelation)
+
+	// Initialize the input components
+	// in preparation for the main run loop.
+	err = in.setup(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Start the main run loop
+	err = in.run(ctx)
+	if err != nil {
+		in.log.Errorw("error running input", "error", err)
+		return err
+	}
+
+	return nil
+}
+
+// setup initializes the input components.
+//
+// The main components are:
+// 1. Azure Storage Leaser / Checkpointer
+// 2. Event Processor Host
+// 3. Message handler
```

Comment:
The comments in the whole PR, especially in this function, are very helpful :+1: